src/main/kotlin/earth/terrarium/techarium/common/blocks/base/DirectionalDeployableMachine.kt

```diff
@@ -0,0 +1,45 @@
+package earth.terrarium.techarium.common.blocks.base
+
+import earth.terrarium.techarium.common.util.toRotation
+import net.minecraft.core.BlockPos
+import net.minecraft.core.Direction
+import net.minecraft.world.item.context.BlockPlaceContext
+import net.minecraft.world.level.LevelAccessor
+import net.minecraft.world.level.block.Block
+import net.minecraft.world.level.block.HorizontalDirectionalBlock
+import net.minecraft.world.level.block.Mirror
+import net.minecraft.world.level.block.Rotation
+import net.minecraft.world.level.block.state.BlockState
+import net.minecraft.world.level.block.state.StateDefinition
+
+abstract class DirectionalDeployableMachine(properties: Properties): DeployableMachine(properties) {
+    companion object{
+        val FACING = HorizontalDirectionalBlock.FACING
+    }
+
+    init {
+        registerDefaultState(defaultBlockState()
+            .setValue(FACING, Direction.NORTH))
+    }
+
+    override fun createBlockStateDefinition(builder: StateDefinition.Builder<Block, BlockState>) {
+        super.createBlockStateDefinition(builder)
+        builder.add(FACING)
+    }
+
+    override fun rotate(state: BlockState, level: LevelAccessor, pos: BlockPos, direction: Rotation): BlockState {
```

Comment:
Maybe a nitpick, but these can be expression functions

Reply:
you are right I will change them